tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/EmailSend.scala

```diff
@@ -0,0 +1,270 @@
+package com.linagora.tmail.james.jmap.model
+
+import cats.implicits.toTraverseOps
+import com.linagora.tmail.james.jmap.json.EmailSendSerializer
+import com.linagora.tmail.james.jmap.model.EmailSendCreationRequestRaw.{emailCreateAssignableProperties, emailSubmissionAssignableProperties}
+import eu.timepit.refined.auto._
+import eu.timepit.refined.refineV
+import org.apache.james.jmap.core.Id.{Id, IdConstraint}
+import org.apache.james.jmap.core.Properties.toProperties
+import org.apache.james.jmap.core.SetError.SetErrorDescription
+import org.apache.james.jmap.core.{AccountId, Id, Properties, SetError, UTCDate, UuidState}
+import org.apache.james.jmap.json.EmailSetSerializer
+import org.apache.james.jmap.mail.Email.Size
+import org.apache.james.jmap.mail.{BlobId, DestroyIds, EmailCreationRequest, EmailCreationResponse, EmailSet, EmailSetRequest, EmailSubmissionId, Envelope, ThreadId, UnparsedMessageId}
+import org.apache.james.jmap.method.WithAccountId
+import org.apache.james.mailbox.MessageManager.AppendCommand
+import org.apache.james.mailbox.model.MessageId
+import org.apache.james.mime4j.dom.Message
+import play.api.libs.json.{JsError, JsObject, JsPath, JsSuccess, JsonValidationError}
+
+import java.time.ZonedDateTime
+import java.util.{Date, UUID}
+import javax.mail.Flags
+
+object EmailCreationRequest {
+  def asAppendCommand(request: EmailCreationRequest, message: Message): AppendCommand =
```

Comment:
Couldn't it be moved to the EmailCreationRequest instead to avoid passing the EmailCreationRequest argument?

Ideally:

```
request.asAppendCommand()
```

Reply:
I have a bit afraid to modify the code in the `james-project` when I do Tmail

Reply:
I have a bit afraid to modify the code in the james-project when I do Tmail

Reply:
This is just a suggestion and the added value here is minimal.

But you should not.

If we can make some refactoring on James side that makes James code more beautiful, then let's call it a refactoring and for sure we can contribute it :-)